//-------------------------------------------------------------------------------
// Name:        navigation_data.proto
// Purpose:     protobuf and grpc for remote control of navigation server
//
// Author:      Laurent Carré
//
// Created:     16/07/2024
// Copyright:   (c) Laurent Carré Sterwen Technology 2021-2025
// Licence:     Eclipse Public License 2.0
//-------------------------------------------------------------------------------

syntax = "proto3";

enum EngineState {
    ENGINE_OFF=0; // No power
    ENGINE_ON=1;  // Powered on but not running
    ENGINE_RUNNING=2;
  }

message engine_data {
  uint32 engine_id=1;
  EngineState state=2;
  float total_hours=3;
  float speed=4;
  float temperature=5;
  float alternator_voltage=8;
  string last_start_time=6; // ISO format
  string last_stop_time=7;
}

message engine_response {
    uint32 id=4;
    string error_message=1;
    engine_data data=2;
    repeated engine_event events=3;
}

message engine_request {
  uint32 id = 2;
  uint32 engine_id=1;
}

message engine_event {
  uint32 engine_id=1;
  string timestamp=2;
  float total_hours=3;
  EngineState current_state=4;
  EngineState previous_state=5;
}

service EngineData {
  rpc GetEngineData(engine_request) returns(engine_response) {}
  rpc GetEngineEvents(engine_request) returns (engine_response) {}
}